### PR TITLE
Clean up sign up spec & add start of the style guide

### PR DESCRIPTION
**Pull request by alisterscott**
Cleans up the sign up spec

Adds a domain registration details test data helper

Adds the beginning of a style guide for writing tests

Adds filling in the "about" page in one of the sign up scenarios (we didn't ever do this previously)

Includes some parts of #923

### CODE REVIEW

**Comment by alisterscott** [comment]
@Automattic/flowpatrol this is a HUGE PR, but I wanted to clean things up in this spec, hopefully it will make the sign up tests faster and determistic, and also easier to read.

I've started a style guide as recommended by @brbrr  in #923

File: lib/pages/signup/about-page.js [hunk 1]
```diff
@@ -9,6 +9,26 @@ export default class AboutPage extends BaseContainer {
 	}
 
 	submitForm() {
-		return driverHelper.clickWhenClickable( this.driver, By.css( '.about__submit-wrapper button.is-primary' ), this.explicitWaitMS );
+		return driverHelper.clickWhenClickable( this.driver, By.css( '.about__submit-wrapper button.is-primary' ) );
+	}
+
+	enterSiteDetails( siteTile, siteTopic, { showcase = false, share = false, sell = false, educate = false, promote = false } = {} ) {
```
Comment (rachelmcr) [inline lib/pages/signup/about-page.js]: Small typo in this and the next line: `SiteTile` -> `SiteTitle`.

File: specs/wp-signup-spec.js [hunk 2]
```diff
@@ -271,96 +229,86 @@ testDescribe( `[${host}] Sign Up  (${screenSize}, ${locale})`, function() {
...
 		} );
 
 		test.describe( `Step ${stepNum}: About Page`, function() {
 			stepNum++;
 
 			test.it( 'Can see the about page', function() {
-				this.startPage = new StartPage( driver, { visit: true, culture: locale } );
-				this.aboutPage = new AboutPage( driver );
-				return this.aboutPage.displayed().then( ( displayed ) => {
+				new StartPage( driver, { visit: true, culture: locale } ).displayed();
```
Comment (rachelmcr) [inline specs/wp-signup-spec.js]: Should we also break this out into its own step, as in the previous test (lines 87-95)?

File: specs/wp-signup-spec.js [hunk 3]
```diff
@@ -505,38 +431,27 @@ testDescribe( `[${host}] Sign Up  (${screenSize}, ${locale})`, function() {
 								} );
 							} );
 
-							test.it( 'The sign up processing page will finish and show a \'Continue\' button', function() {
-								return this.signupProcessingPage.waitForContinueButtonToBeEnabled();
-							} );
-
-							test.it( 'Clicking the \'Continue\' button continues the process', function() {
-								return this.signupProcessingPage.continueAlong();
+							test.it( 'Can then see the sign up processing page and it will finish and show a \'Continue\' button, which is clicked', function() {
```
Comment (rachelmcr) [inline specs/wp-signup-spec.js]: With this change it looks like you can remove lines 427-432 above (starting with `test.it( 'Can then see the sign up processing page', function() {` ).

**Comment by rachelmcr** [comment]
The style guide looks great! I started to review the signup spec but only got about halfway through. I skimmed through the rest of the changes but it would be great for another reviewer to take a look at this, too.

File: lib/pages/signup/about-page.js (hunk 1 above)
Comment (alisterscott) [reply lib/pages/signup/about-page.js]: Thanks 👍

File: specs/wp-signup-spec.js (hunk 2 above)
Comment (alisterscott) [reply specs/wp-signup-spec.js]: Good idea
I didn't enter any of the "about" information in this scenario, I thought entering a random site title could cause the visual difference tests to complain about the dynamic information? What do you think? Just submit that form as we did before?

File: specs/wp-signup-spec.js (hunk 3 above)
Comment (alisterscott) [reply specs/wp-signup-spec.js]: Yes, good catch 👍

**Comment by alisterscott** [comment]
@brbrr I'm keen to get this merged today - to avoid conflicts - would you have time to add additional review on @rachelmcr's above? It's a big PR but it kinda needs to fit together.

File: specs/wp-signup-spec.js [hunk 4]
```diff
@@ -271,96 +229,89 @@ testDescribe( `[${host}] Sign Up  (${screenSize}, ${locale})`, function() {
...
+						return pickAPlanPage.displayed().then( ( displayed ) => {
 							eyesHelper.eyesScreenshot( driver, eyes, 'Plans Page' );
 							return assert.equal( displayed, true, 'The pick a plan page is not displayed' );
 						} );
 					} );
 
 					test.it( 'Can select the premium plan', function() {
-						return this.pickAPlanPage.selectPremiumPlan();
+						const pickAPlanPage = new PickAPlanPage( driver );
+						return pickAPlanPage.selectPremiumPlan();
```
Comment (brbrr) [inline specs/wp-signup-spec.js]: Can we combine this step and step above? This should become a bit shorter

**Comment by alisterscott** [comment]
@brbrr 

> test.describe( 'Step ${stepNum}: ....

Just a way to print step numbers like Step 1 - Domains, Step 2 Plans, Step 3 Checkout etc

We used to write them out like that but the steps constantly reorder and get out of numbered sync - so we put them dynamically

Any ideas to improve?

File: specs/wp-signup-spec.js (hunk 4 above)
Comment (alisterscott) [reply specs/wp-signup-spec.js]: Good idea - thanks

**Comment by brbrr** [comment]
When I run these tests locally - these `describe` blocks did not help me in debugging, they are just messing the output.

I would get rid of these blocks completely. We don't have them anywhere in other test files, and pattern that we use, 1 `describe` block per test, works well for me.

as an improvement, we could also try to combine some of the common actions into flow file.

**Comment by alisterscott** [comment]
> When I run these tests locally - these describe blocks did not help me in debugging, they are just messing the output.
> I would get rid of these blocks completely. We don't have them anywhere in other test files, and pattern that we use, 1 describe block per test, works well for me.

Yes, legacy code before we could bail, so we had to nest

> as an improvement, we could also try to combine some of the common actions into flow file.

Yes, we have some of that already but would be worth trialling - separate PR though 😉

**Comment by alisterscott** [comment]
Merging this while it's all good ✅

**Comment by alisterscott** [comment]
Thanks for all your feedback @rachelmcr and @brbrr

File: specs/wp-signup-spec.js (hunk 2 above)
Comment (rachelmcr) [reply specs/wp-signup-spec.js]: Yes, agreed, I wouldn't enter any "About" information here. If we wanted to enter information I'd do it after the screenshot is taken for the visual diff test, but also I think it's worth having a test that verifies you can continue without entering information here.

**Pull request merged**